geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -34,10 +34,27 @@ object FilterHelper {
   def updateTopologicalFilters(filter: Filter, sft: SimpleFeatureType): Filter =
     filter.accept(new SafeTopologicalFilterVisitorImpl(sft), null).asInstanceOf[Filter]
 
+  def getFirstBinarySpatialOpPropertyName(op: BinarySpatialOperator): PropertyName = op.getExpression1 match {
+    case pn: PropertyName => pn
+    case _                => op.getExpression2 match {
+      case pn: PropertyName => pn
+      case _                =>
+        throw new Exception(s"Neither child of a binary spatial operator was a property name:  $op")
+    }
+  }
+
+  def getFirstBinarySpatialOpPropertyGeometry(op: BinarySpatialOperator): Geometry = op.getExpression1 match {
+    case lit: Literal => lit.evaluate(null, classOf[Geometry])
+    case _            => op.getExpression2 match {
+      case lit: Literal => lit.evaluate(null, classOf[Geometry])
+      case _            =>
+        throw new Exception(s"Neither child of a binary spatial operator was a literal geometry:  $op")
+    }
+  }
```

Comment:
This is good for now. We have some code that does something similar here: org.locationtech.geomesa.filter#checkOrder. When we go back to evaluate GEOMESA-1155 we can revisit.